fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java

```diff
@@ -660,26 +658,8 @@ public List<Long> getWorkersByWorkerGroup(long workerGroupId) throws UserExcepti
             List<WorkerGroupDetailInfo> workerGroupDetailInfos = client.
                     listWorkerGroup(serviceId, Collections.singletonList(workerGroupId), true);
             for (WorkerGroupDetailInfo detailInfo : workerGroupDetailInfos) {
-                List<WorkerInfo> workerInfos = detailInfo.getWorkersInfoList();
-                for (WorkerInfo workerInfo : workerInfos) {
-                    if (workerToBackend.containsKey(workerInfo.getWorkerId())) {
-                        nodeIds.add(workerToBackend.get(workerInfo.getWorkerId()));
-                    } else {
-                        // workerToBackend may not container this worker, so need to get it from systemInfoSerivce
-                        // and fill it
-                        long workerId = workerInfo.getWorkerId();
-                        String workerAddr = workerInfo.getIpPort();
-                        String[] pair = workerAddr.split(":");
-                        long nodeId = getAvailableBackendId(pair[0], Integer.parseInt(pair[1]));
-                        if (nodeId != -1L) {
-                            nodeIds.add(nodeId);
-
-                            // put it into map
-                            workerToId.put(workerAddr, workerId);
-                            workerToBackend.put(workerId, nodeId);
-                        }
-                    }
-                }
+                detailInfo.getWorkersInfoList()
+                        .forEach(x -> getOrUpdateBackendIdByWorkerInfo(x).ifPresent(nodeIds::add));
```

Comment:
add a lock in getOrUpdateBackendIdByWorkerInfo, since it's ReentrantReadWriteLock, here is no lock protected

Reply:
done